edb/server/protocol/binary.pyx

```diff
@@ -366,6 +366,7 @@ cdef class EdgeConnection:
 
     cdef abort_pinned_pgcon(self):
         if self._pinned_pgcon is not None:
+            self._pinned_pgcon.pinned_by = None
             self._pinned_pgcon.abort()
```

Comment:
Should this be moved directly into `PgCon.abort()`? Also, how did you stumble upon this?

Reply:
> Should this be moved directly into `PgCon.abort()`?

Negative. In `PgCon.fallthrough_idle()`, we use `PgCon.abort()` to send `aborted_with_error` to EdgeDB clients.

> Also, how did you stumble upon this?

I was in a transaction in the CLI when quitting the CLI, and the server complained releasing the same pgcon twice. Then it's like `EdgeCon.abort_pinned_pgcon()` and `EdgeCon.on_aborted_pgcon() ` are both called.

Reply:
> I was in a transaction in the CLI when quitting the CLI, and the server complained releasing the same pgcon twice. Then it's like EdgeCon.abort_pinned_pgcon() and EdgeCon.on_aborted_pgcon() are both called.

Can you replicate this in a test?

Reply:
Sure, let me try.